service/collector/trusted_advisor.go

```diff
@@ -237,17 +358,13 @@ func (t *TrustedAdvisor) getTrustedAdvisorResources(id *string, awsClients aws.C
 	return checkResultOutput.Result.FlaggedResources, nil
 }
 
-func resourceToMetrics(resource *support.TrustedAdvisorResourceDetail, accountID string) (prometheus.Metric, prometheus.Metric, error) {
-	if len(resource.Metadata) != resourceMetadataLength {
-		return nil, nil, invalidResourceError
-	}
-
-	region := resource.Metadata[indexRegion]
-	service := resource.Metadata[indexService]
-	limitName := resource.Metadata[indexName]
+func resourceToMetrics(resource trustedAdvisorResource, accountID string) (prometheus.Metric, prometheus.Metric, error) {
+	region := resource.Region
```

Comment:
I think you might need to remove the pointer in line 387 and below

Reply:
Makes sense!